core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -476,15 +476,17 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
-  private def withClonedRealm(sourceWorkspaceContext: SlickWorkspaceContext, destWorkspaceRequest: WorkspaceRequest)(op: (Option[ManagedGroupRef]) => ReadWriteAction[PerRequestMessage]): ReadWriteAction[PerRequestMessage] = {
-    // if the source has a realm, the dest must also have that realm or no realm, and the source realm is applied to the destination
-    // otherwise, the caller may choose to apply a realm
-    (sourceWorkspaceContext.workspace.authorizationDomain, destWorkspaceRequest.authorizationDomain) match {
-      case (Some(sourceRealm), Some(destRealm)) if sourceRealm != destRealm =>
-        val errorMsg = s"Source workspace ${sourceWorkspaceContext.workspace.briefName} has realm $sourceRealm; cannot change it to $destRealm when cloning"
-        DBIO.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.UnprocessableEntity, errorMsg)))
-      case (Some(sourceRealm), _) => op(Option(sourceRealm))
-      case (None, destOpt) => op(destOpt)
+  private def withClonedAuthDomain(sourceWorkspaceContext: SlickWorkspaceContext, destWorkspaceRequest: WorkspaceRequest)(op: (Set[ManagedGroupRef]) => ReadWriteAction[PerRequestMessage]): ReadWriteAction[PerRequestMessage] = {
+    // if the source has an auth domain, the dest must also have that auth domain as a subset
+    // otherwise, the caller may choose to add to the auth domain
+    val sourceWorkspaceADs = sourceWorkspaceContext.workspace.authorizationDomain
+    val destWorkspaceADs = destWorkspaceRequest.authorizationDomain
+
+    if(sourceWorkspaceADs.subsetOf(destWorkspaceADs)) op(sourceWorkspaceADs ++ destWorkspaceADs)
+    else {
+      val missingGroups = sourceWorkspaceADs -- destWorkspaceADs
+      val errorMsg = s"Source workspace ${sourceWorkspaceContext.workspace.briefName} has an Authorization Domain containing the groups $missingGroups which, which are missing on the destination workspace"
```

Comment:
Think there's a typo in this error message. I believe it should read "...containing the groups $missingGroups, which are missing..." (in other words, just delete the first "which")